File: SDKs/amazonaws.com/textract/2018-06-27/python/src/sdk/models/shared/expensedetection.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class ExpenseDetection:
    r"""ExpenseDetection
    An object used to store information about the Value or Label detected by Amazon Textract.
    """
    
    confidence: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Confidence') }})
    geometry: Optional[Geometry] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Geometry') }})
    text: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Text') }})
    

File: SDKs/amazonaws.com/textract/2018-06-27/typescript/src/sdk/models/shared/expensedetection.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { Geometry } from "./geometry";



// ExpenseDetection
/** 
 * An object used to store information about the Value or Label detected by Amazon Textract.
**/
export class ExpenseDetection extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=Confidence" })
  confidence?: number;

  @SpeakeasyMetadata({ data: "json, name=Geometry" })
  geometry?: Geometry;

  @SpeakeasyMetadata({ data: "json, name=Text" })
  text?: string;
}
